jobbergate-api/jobbergate_api/apps/applications/schemas.py

```diff
@@ -95,13 +96,30 @@ class JobbergateConfig(BaseModel):
     Model for Jobbergate configuration (subsection at the yaml file).
     """
 
-    default_template: str
+    default_template: Optional[str]
     supporting_files: Optional[List[str]]
     supporting_files_output_name: Optional[Dict[str, List[str]]]
     template_files: Optional[List[str]]
     job_script_name: Optional[str]
     output_directory: str = "."
 
+    @root_validator(pre=True)
+    def compute_extra_settings(cls, values):
+        """
+        Compute missing values and extra operations to enhance the user experience.
+        """
+        # Transform string to list of strings for a better user experience
+        if values.get("supporting_files_output_name"):
+            for k, v in values["supporting_files_output_name"].items():
+                if isinstance(v, str):
+                    values["supporting_files_output_name"][k] = [v]
+
+        # Get the list of supporting files automatically
+        if values.get("supporting_files_output_name") and not values.get("supporting_files"):
```

Comment:
So, the `supporting_files` field will end up being a copy of the keys in the `supporting_files_output_name` if it is not defined? What happens when it *is* defined but it doesn't agree with the keys of `supporting_files_output_name`?

We will need to address the weirdness of these two fields in the future for sure.

Reply:
Yes, `supporting_files` is now optional and can be obtained as the keys in the `supporting_files_output_name`.

For now, just the schema of the config file is checked. We should add business rules in the near future to address this and some of the other fields.